File: app/src/main/java/com/boymask/myapplication/retrofit/OpenAIApi.java
package com.boymask.myapplication.retrofit;

import okhttp3.MultipartBody;
import okhttp3.RequestBody;
import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.*;

public interface OpenAIApi {

    @Multipart
    @POST("v1/files")
    Call<ResponseBody> uploadFile(
            @Header("Authorization") String authorization,
            @Part MultipartBody.Part file,
            @Part("purpose") RequestBody purpose
    );

    @POST("v1/responses")
    Call<ResponseBody> analyzeFile(
            @Header("Authorization") String authorization,
            @Body RequestBody body
    );
}

File: app/src/main/java/com/boymask/myapplication/retrofit/OpenAIApi.kt
package com.boymask.myapplication.retrofit

import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.Headers
import retrofit2.http.POST


interface OpenAIApi {
    @Headers(
        "Content-Type: application/json", //
        "Authorization: Bearer YOUR_API_KEY"
    )
    @POST("v1/chat/completions")
    fun chat(@Body request: ChatRequest?): Call<ChatResponse?>?
}